### PR TITLE
Support deleting selected layer(s) when pen tool is activated

### PR DESCRIPTION
**Pull request by shaoshing**
This PR addresses the following issues:

* #1422  Unable to delete layers if Pen Tool is selected
* #3010 Create new layer by using the pen tool result in error

### CODE REVIEW

**Comment by iwehrman** [comment]
Needs more rebase.

File: src/js/actions/layers.js [hunk 1]
```diff
@@ -2446,7 +2446,15 @@ define(function (require, exports) {
...
                         this.flux.actions.layers.resetLayers(currentDocument, curLayer, true);
                     } else {
-                        this.flux.actions.layers.addLayers(currentDocument, event.layerID);
+                        // The `addLayers` actions is delayed because the new layer is not always avaialble 
+                        // for read when we receive the `make` event. Without the delay, we may result in 
+                        // "base element not found" error. For example, creating a new layer with the pen tool.
+                        // 
+                        // FIXME: Photoshop should make sure the new layer is ready for read before 
+                        // emitting the `make` event.
+                        window.setTimeout(function () {
```
Comment (iwehrman) [inline src/js/actions/layers.js]: Bummer. Can you file a Watson bug for this? And, ideally, reference that bug number here?

**Comment by iwehrman** [comment]
Besides the rebase and the Watson bug, looks good.

File: src/js/actions/layers.js (hunk 1 above)
Comment (shaoshing) [inline src/js/actions/layers.js]: I created a watson bug for this https://watsonexp.corp.adobe.com/#bug=4077564. also updated to issue #3010

**Comment by shaoshing** [comment]
@iwehrman just rebased and created watson bug. Back to you.

**Comment by iwehrman** [comment]
LGTM.

**Pull request merged**